apps/ai-content-generator/src/locations/Dialog.tsx

```diff
@@ -26,7 +26,7 @@ const Dialog = () => {
   };
 
   return (
-    <GeneratorProvider entryId={entryId} prompt={featureConfig[feature].prompt}>
+    <GeneratorProvider entryId={entryId} feature={feature} prompt={featureConfig[feature].prompt}>
```

Comment:
Since we're only using the `feature` in the provider to check for a bool, we could follow a similar pattern to the `isTranslate` prop on CommonGenerator

So instead of getting feature from context, we can have a flag like:
```
case : AIFeature.Title:
  return <CommonGenerator fromPrompt />
```
and then you can pass down that `FromPrompt` into `NewOrExistingText`

Reply:
this is a great point! I felt like what I was doing was a little bit more of a heavy lift than necessary. I ended up implementing you suggestion, with a small change in the prop name to be `isTitle` to keep consistency with the `isTranslate` pattern.